proxy/client_hello.go

```diff
@@ -0,0 +1,48 @@
+package proxy
+
+import (
+	"encoding/binary"
+	"io"
+)
+
+const headerLen = 5
+
+type ClientHello struct {
+	Header     ClientHelloHeader
+	Raw        []byte //Header + Payload
+	RawHeader  []byte
+	RawPayload []byte
+}
+
+type ClientHelloHeader struct {
+	Type         byte
+	ProtoVersion uint16
+	PayloadLen   uint16
+}
+
+func ReadClientHello(r io.Reader) (*ClientHello, error) {
+	var rawHeader [5]byte
```

Comment:
see https://datatracker.ietf.org/doc/html/rfc8446#section-5.2 and https://datatracker.ietf.org/doc/html/rfc6101#section-5.2.1